### PR TITLE
v3: Sync Client to OpenAPI Spec, Bump Ruby Version

### PR DESCRIPTION
**Pull request by joshmenden**
A few big changes in this PR:
- Update the client to be in line with the API's [open_api spec](https://api.ynab.com/papi/open_api_spec.yaml).
- Use v7 of the [OpenAPI Client Generator](https://openapi-generator.tech/)
- Bump the gem's ruby version to 3.3.1, since 2.7 has been [end-of-life since last March](https://www.ruby-lang.org/en/downloads/branches/).

This brings about a few updates to the client, like returning a flag name and the ability to update payees and categories. See the [changelog](https://api.ynab.com/#v1.69.0) for more information.

With this in mind, I decided to go ahead and bump the library to a major breaking change of 3.0.0, since getting the client in sync with the open_api spec required it to remove some previously deprecated schemas.

### CODE REVIEW

File: ynab.gemspec [hunk 1]
```diff
@@ -25,7 +25,8 @@ Gem::Specification.new do |s|
   s.summary     = "YNAB API Endpoints Ruby Gem"
   s.description = "Ruby gem wrapper for the YNAB API. Read the documentation at https://api.ynab.com"
   s.license     = "Apache-2.0"
-  s.required_ruby_version = ">= 2.7"
+  s.required_ruby_version = ">= 3.3"
```
Comment (bradymholt) [inline ynab.gemspec:28]: Do you think it'd be worth adding a note in the README letting users know they can use 2.x version of this client if they are still on Ruby 2?

File: ynab.gemspec [hunk 2]
```diff
@@ -5,10 +5,10 @@
 
 #Our API uses a REST based design, leverages the JSON data format, and relies upon HTTPS for transport. We respond with meaningful HTTP response codes and if an error occurs, we include error details in the response body.  API Documentation is at https://api.ynab.com
 
-The version of the OpenAPI document: 1.0.0
+The version of the OpenAPI document: 1.69.0
 
 Generated by: https://openapi-generator.tech
-OpenAPI Generator version: 6.6.0
+Generator version: 7.6.0
```
Comment (bradymholt) [inline ynab.gemspec:11]: Did you install `openapi-generator` with brew? I think I overlooked documenting this needed step in https://github.com/ynab/ynab-sdk-ruby/blob/main/DEVELOPMENT.md.  Can we add a mention for this there, to install and/or update `openapi-generator`?

File: lib/ynab/models/user_response_data.rb [hunk 3]
```diff
@@ -100,37 +100,30 @@ def hash
     # @param [Hash] attributes Model attributes in the form of hash
     # @return [Object] Returns the model itself
     def self.build_from_hash(attributes)
-      new.build_from_hash(attributes)
```
Comment (bradymholt) [inline lib/ynab/models/user_response_data.rb:103]: Are all these changes coming directly from the generator?  Were any of them made manually?

**Review by bradymholt — APPROVED** [review]
Changes look good!  I had a couple of questions.  Do all the rspec tests pass on this branch locally?  For some reason the GitHub Actions build did not run here; possibly because this PR source is a fork.

File: ynab.gemspec (hunk 1 above)
Comment (joshmenden) [reply ynab.gemspec:28]: Yes, I think that's a good idea. Done in 2e5e58ae189f36018760d9d43d5c27bd7286b759

File: ynab.gemspec (hunk 2 above)
Comment (joshmenden) [reply ynab.gemspec:11]: Sure thing, done with 2fcf5135f13720c37127194fb7a7f3d33550f5b8

**Comment by bradymholt** [comment]
:shipit:

**Comment by bradymholt** [comment]
Tests running with these changes on `main` branch: https://github.com/ynab/ynab-sdk-ruby/actions/runs/9212146210

**Pull request merged**